docs/learning/WriteAConnector.md

```diff
@@ -1,108 +1,727 @@
+
+- BridgeRunner: [Opens a local briefcase copy](https://github.com/imodeljs/imodeljs/tree/master/docs/learning/backend/IModelDb.md) of the iModel that is to be updated.
+- Import or Update Schema
+  - Connector: Possibly [import an appropriate BIS schema into the briefcase](https://github.com/imodeljs/imodeljs/tree/master/docs/learning/backend/SchemasAndElementsInTypeScript.md#importing-the-schema) or upgrade an existing schema.
+  - BridgeRunner: [Push](https://github.com/imodeljs/imodeljs/tree/master/docs/learning/backend/IModelDbReadwrite.md#pushing-changes-to-imodelhub) the results to the iModelServer.
+- Convert Changed Data
+  - Connector:
+    - Opens to the data source.
+    - Detect changes to the source data.
+    - [Transform](#data-alignment) the new or changed source data into the target BIS schema.
+    - Write the resulting BIS data to the local briefcase.
+    - Remove BIS data corresponding to deleted source data.
+  - BridgeRunner: Obtain required [Locks and Codes](https://github.com/imodeljs/imodeljs/tree/master/docs/learning/backend/ConcurrencyControl.md) from the iModel server and code server.
+- BridgeRunner: [Push](https://github.com/imodeljs/imodeljs/tree/master/docs/learning/backend/IModelDbReadwrite.md#pushing-changes-to-imodelhub) changes to the iModel server.
+
+### Analyzing the Connector output
+
+As a Connector developer, once the data is transformed into an iModel, one needs tools to analyze the validity of the conversion. The sections below give a summary of the recommended tools which allow a developer to query the graphical and non-graphical data in an iModel.
+
+#### ECSQL
+
+Please see this article on [ECSQL](https://www.itwinjs.org/learning/ecsqltutorial/) as a prerequisite for this section. The iModelConsole instance below demonstrates how you can use ECSQL to find some of the transformed data in an iModel.
+
+<figure>
+  <iframe style="height:40vh;width:60vw;" src="https://www.itwinjs.org/console/?imodel=House%20Sample%20Bak&query=SELECT%20Element.Id%20FROM%20bis.ExternalSourceAspect%20WHERE%20Identifier=%27197359%27">
+  </iframe>
+</figure>
+
+Some sample queries that is helpful to debug Connector output
+
+1. Get the iModel Element id from the identifier from a given source file
+
+   ```sql
+    SELECT Element.Id FROM bis.ExternalSourceAspect WHERE Identifier='<Element Id from File>'
+   ```
+
+2. List of all known repositories (links to files in an iModel)
+
+   ```sql
+   SELECT UserLabel, Url FROM bis.repositoryLink
+   ```
+
+3. List of all physical partitions present in the iModel
+
+   ```sql
+   SELECT ECInstanceId, CodeValue from bis.physicalpartition
+   ```
+
+#### Visualizing the output
+
+To get started, please build it using the instructions provided in the [display test app README](https://github.com/imodeljs/imodeljs/blob/master/test-apps/display-test-app/README.md)

```

Comment:
We do not want people cloning and using the Display Test App outside of someone working directly on iTwin.js in the repo.

I'd suggest you point people to the iTwin Desktop Viewer as the way they look at their local data run through the Connector.  There is a tutorial here, https://www.itwinjs.org/learning/tutorials/develop-desktop-viewer/.

Side note: That tutorial needs to be updated, working on that today so hopefully will be in place...